packaging/nix/wrapper

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+# This file is copied from https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/clang-tools/wrapper
+# The clang-tools wrapper is commonly used together with the clang package on
+# nix, because without the wrapper, clang tools fail to find stdlib includes on
+# nix.
+
+buildcpath() {
+  local path after
+  while (( $# )); do
+    case $1 in
+        -isystem)
+            shift
+            path=$path${path:+':'}$1
+            ;;
+        -idirafter)
+            shift
+            after=$after${after:+':'}$1
+            ;;
+    esac
+    shift
+  done
+  echo $path${after:+':'}$after
+}
+
+export CPATH=${CPATH}${CPATH:+':'}$(buildcpath ${NIX_CFLAGS_COMPILE} \
+                                               $(<@clang@/nix-support/libc-cflags)):@clang@/resource-root/include
+export CPLUS_INCLUDE_PATH=${CPLUS_INCLUDE_PATH}${CPLUS_INCLUDE_PATH:+':'}$(buildcpath ${NIX_CFLAGS_COMPILE} \
+                                                                                      $(<@clang@/nix-support/libcxx-cxxflags) \
+                                                                                      $(<@clang@/nix-support/libc-cflags)):@clang@/resource-root/include
```

Comment:
I would be actually surprised if you don't need these headers for anything libclang based. Usually if you use a compiler on Nix, we have in nixpkgs a wrapper around c/c++ compiler that add the same paths. I can trigger issues with includes as simple as `#include <string>`.

Reply:
@Mic92 As you can see in the above discussion in the PR it seems to work perfectly fine without these changes - both in `nix` and `NixOS`. I'm still trying to understand what problem this PR solves, but unfortunately I have a very minimal experience with `nix`...

It would help if I had exact procedure to reproduce the issue.

Reply:
I was just trying to reproduce your example, but I cannot reproduce it as.
If I rust `../../result/bin/clang-uml --progress -g mermaid` instead `hatchcare/libs` it just does nothing:

```
newfstatat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/clang-uml/hatchcare/diagrams", {st_mode=S_IFDIR|0755, st_size=2, ...}, 0) = 0
newfstatat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/clang-uml/hatchcare/diagrams", {st_mode=S_IFDIR|0755, st_size=2, ...}, 0) = 0
newfstatat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/clang-uml/hatchcare/diagrams", {st_mode=S_IFDIR|0755, st_size=2, ...}, 0) = 0
openat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/clang-uml/hatchcare/libs/build/compile_flags.txt", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/clang-uml/hatchcare/libs/build/compile_commands.json", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/clang-uml/hatchcare/libs/build/compile_flags.txt", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/clang-uml/hatchcare/libs/build/compile_commands.json", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/clang-uml/hatchcare/libs/compile_flags.txt", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/clang-uml/hatchcare/libs/compile_commands.json", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/clang-uml/hatchcare/compile_flags.txt", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/clang-uml/hatchcare/compile_commands.json", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/clang-uml/compile_flags.txt", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/clang-uml/compile_commands.json", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/compile_flags.txt", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/epic-neumann-8iqn3f/compile_commands.json", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/compile_flags.txt", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/compile_commands.json", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/compile_flags.txt", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/compile_commands.json", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
futex(0x7fffeb467070, FUTEX_WAKE_PRIVATE, 2147483647) = 0
gettid()                                = 1487205
exit_group(1)                           = ?
+++ exited with 1 +++
```

```
[joerg@turingmachine libs]$ ls -la /tmp/epic-neumann-8iqn3f/clang-uml/hatchcare/diagrams
total 17
drwxr-xr-x  2 joerg users  2 May 30 18:05 .
drwxr-xr-x 18 joerg users 35 May 30 18:05 ..
```

Reply:
This was on my NixOS. I will now try the nix container for completness.

Reply:
@Mic92 The problem above seems to be missing `compile_commands.json` file, it should be in `hatchcare/libs/build/compile_commands.json`

Reply:
Same problem happens in the docker container, after running your commands without the second "nix develop" that doesn't because there is no flake.nix:

```
bash-5.2# ls -la ../../../hatchcare/diagrams/hatchcare_*
-rw-r--r-- 1 root root  12339 May 30 16:14 ../../../hatchcare/diagrams/hatchcare_cpp.puml
-rw-r--r-- 1 root root 103243 May 30 16:14 ../../../hatchcare/diagrams/hatchcare_cpp.svg
-rw-r--r-- 1 root root  17031 May 30 16:14 ../../../hatchcare/diagrams/hatchcare_dart.puml
-rw-r--r-- 1 root root 151831 May 30 16:14 ../../../hatchcare/diagrams/hatchcare_dart.svg
-rw-r--r-- 1 root root  26765 May 30 16:14 ../../../hatchcare/diagrams/hatchcare_dart.uml
-rw-r--r-- 1 root root  11651 May 30 16:14 ../../../hatchcare/diagrams/hatchcare_wrapper.puml
-rw-r--r-- 1 root root  95732 May 30 16:14 ../../../hatchcare/diagrams/hatchcare_wrapper.svg
bash-5.2# rm ../../../hatchcare/diagrams/hatchcare_*
bash-5.2# /tmp/clang-uml/result/bin/clang-uml --progress -g mermaid
bash-5.2# ls -la ../../../hatchcare/diagrams/hatchcare_*
ls: cannot access '../../../hatchcare/diagrams/hatchcare_*': No such file or directory

```

Reply:
Is this maybe a very recent regression?

Reply:
@Mic92 Sorry, I don't understand what you mean here. Can you paste all commands in order you've typed them, and also paste output of `pwd` and `ls build/compile_commands.json` just before running `clang-uml`, because the commands posted by @hatch01 in the PR work in both Nix Docker container and in a clean NixOS VM - I checked yesterday...